### PR TITLE
new class Race

### PR DESCRIPTION
**Pull request by Evilkoy**

### CODE REVIEW

File: src/Race.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+public class Race {
+    private int raceRate = 200;
+
+    public void race(Player player, int upgradeRating) {
+        int pos = getPosition(player, upgradeRating);
+        System.out.println("Your position is " + pos + "/12");
+        player.setMoney(player.getMoney() + (3000 + (3000 / pos)));
+        System.out.println("Your reward: " + (3000 + (3000 / pos)) + "$");
```
Comment (maxxa1234) [inline src/Race.java]: We can make separate var for  3000 + (3000 / pos) because it uses more than one time

File: src/Race.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            raceRate += 30;
+        }
+    }
+
+    private int getPosition(Player player, double upgradeRating) {
+        double rating = player.getMyCar().getRating() + upgradeRating;
+        double pos = 12 * (1 - (rating / raceRate));
+        if (pos < 1) {
+            pos = 1;
+        }
```
Comment (maxxa1234) [inline src/Race.java:19]: What if there will be a few equal places like 1.42, 1.95, 0,45

**Review by maxxa1234 — CHANGES_REQUESTED** [review]

File: Racing_Mx.iml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <module type="JAVA_MODULE" version="4">
-  <component name="NewModuleRootManager" inherit-compiler-output="true">
+  <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_17" inherit-compiler-output="true">
```
Comment (maxxa1234) [inline Racing_Mx.iml]: What is it?

File: src/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
+Main-Class: Main
+
```
Comment (maxxa1234) [inline src/META-INF/MANIFEST.MF:3]: what is it?

File: src/UpgradeShop.java [hunk 5]
```diff
@@ -1,44 +1,43 @@
...
                 System.out.println(exception.getMessage());
                 upgradeChoice(player);
             }
-        } else if (i > 6) {
+        } else if (i > 7) {
             System.out.println("Not correct, make another choice");
             upgradeChoice(player);
+        } else if (i == 6) {
+            Race.race(player, getRatings(player));
         }
```
Comment (maxxa1234) [inline src/UpgradeShop.java:35]: What  if i == 7?

**Review by maxxa1234 — CHANGES_REQUESTED** [review]

File: src/UpgradeShop.java (hunk 5 above)
Comment (Evilkoy) [reply src/UpgradeShop.java:35]: image

**Pull request closed without merging**